cEP-0017.md

```diff
@@ -0,0 +1,176 @@
+bears that have different installation instructions for each distribution such as
+`DartLintBear` and bears that have bear dependencies.
+
+Additionally, the tool should also be able to provide verbose output, ask user for
+permission before proceeding and install bears related to a specific language.
+
+## Proposed Change
+
+### Here is the procedure for the installation of the bears
+
+1. Identify the valid bear names that need to be installed as specified by the
+   user on the command line.
+2. Install bear on the valid bear names. We will run commands from `CondaRequirement`
+   to check whether the package is uploaded to `anaconda.org` and then check whether
+   it can be installed from `PyPi` using `PipRequirement`.
+   - This will require to create conda packages for bears that have different
+     installation instructions for each distribution. This is done by making `bld.bat`
+     file that has installation instruction for Windows and similarly a `bld.sh` file
+     that has installation instruction for Linux.
+3. After installing the bear we proceed to installing its requirements. We call
+   methods from the requirement classes to do this.
+   - Add new Requirement classes to support different dependencies from various linters.
+   - The DistributionRequirement class needs to be split into different classes for each
+     package manager and call its method to introduct more consistency.
+4. We then proceed to install bear dependencies such as for `ClangCloneDetectionBear` by
+   running `install_bears` method on each dependency bear.
+
+### Here is the process for upgrading bears
+
+1. Check the bear name is installed using `PipRequirement` or `CondaRequirement`.
+2. Iterate through `Requirement` instances to make sure that the required version of the
+   linter is installed or upgrade the version if required.
+
+### Split `DistributionRequirement` for each distribution
+
+Specifically this will cover the following platforms:
+
+| Operating System | Tools      |
+| ---------------- | ---------- |
+| Debian           | apt-get    |
+| Ubuntu           | apt-get    |
+| CentOS           | yum        |
+| Fedora           | dnf        |
+| Arch Linux       | pacman     |
+| openSUSE         | zypper     |
+| Void Linux       | xbps       |
+| macOS            | brew       |
+| Windows          | Chocolatey |
+
+### Using features from depman
+
+Depman provides a way to install dependencies by specifying them through `requirements.yml`.
+Currently, only `pip` and `apt-get` are supported but by adding support for new installers we
+would be able to generate a `requirements.yml` specifying the bears and their dependencies.
+It would be useful to have `requirements.yml` in each project for the user that uses coala
+therefore, the cib should provide this additional option.
+
+### Other issues
+
+Allow `generate_package` to generate conda packages and move it from coala-bears to a different
```

Comment:
Where will those conda packages be shared?  Will they be uploaded to some central repo.  Does coala host this conda repo for bears?  Someone else?

move 'it' (the script?) from coala-bears to where?  decision time.
move 'it' why?

Is it going to generate conda packages for any bear, or only bears with python deps?